src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,75 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Class that contains all the tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Deletes the task with index taskNumber.
+     * @param taskNumber the index of task to remove.
+     * @throws DukeException if given task does not exist.
+     */
+    public Task deleteTask(int taskNumber) throws DukeException {
+        if (taskNumber < tasks.size()) {
+            Task toRemove = tasks.get(taskNumber);
+            tasks.remove(taskNumber);
+            return toRemove;
+        } else {
+            throw new DukeException("☹ OOPS!!! The delete command needs to be followed by an existing task number.");
+        }
+    }
+
+    /**
+     * Completes the task with index taskNumber.
```

Comment:
I like your javadocs! They are very clear and easy to understand. I like that you include any issues with the methods too.